cql/src/main/java/org/apache/cassandra/stargate/transport/internal/ServerConnection.java

```diff
@@ -129,8 +130,11 @@ void applyStateTransition(Message.Type requestType, Message.Type responseType) {
               .getAuthenticator()
               .newSaslNegotiator(clientInfo.remoteAddress().getAddress(), certificates());
 
+      // todo how to pass headers here
```

Comment:
@dougwettlaufer 
The `SaslNegotiator` is problematic here because c2 is initializing the negotiator with `orgId` (that needs to be retrieved using `tenant_id`).
Because of that, we need to pass the headers to the `AuthenticationService#getSaslNegotiator` method.
From what I see, the `ServerConnection` is created in the `Server#initChannel`. Is the channel created for every user request? 
- If yes, then maybe we can propagate and extract this information from `ClientInfo clientInfo`?
- If no and the `ServerConnection` is shared between multiple requests, it means that we need to carry the `headers` in the `clientResponse`:
```
 interface SaslNegotiator {
    byte[] evaluateResponse(byte[] clientResponse)
}
```
or modify this method to take headers.
to be able to extract headers for every request.
wdyt?

Reply:
You'll want to sync up with @tjake regarding his change here https://github.com/stargate/stargate/pull/564/files